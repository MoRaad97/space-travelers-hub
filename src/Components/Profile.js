import React from 'react';
import { useSelector } from 'react-redux';
import './style/Profile.css';

const Profile = () => {
  const rockets = useSelector((state) => state.rockets);
  const missions = useSelector((state) => state.mission);

  const revervedRockets = rockets.filter((rocket) => rocket.reserved);
  const joinedMissions = missions.filter((mission) => mission.joined);
  return (
    <section className="my-profile">
      <div className="container">
        <table>
          <thead>My Missions</thead>
          <tbody>
            {joinedMissions.map((element) => (
              <tr
                key={element.mission_id}
                className="profile-row"
              >
                {element.mission_name}
              </tr>
            ))}
          </tbody>
        </table>
        <table>
          <thead>My Rockets</thead>
          <tbody>
            {revervedRockets.map((element) => (
              <tr
                key={element.rocket_id}
                className="profile-row"
              >
                {element.rocket_name}
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    </section>
  );
};

export default Profile;
