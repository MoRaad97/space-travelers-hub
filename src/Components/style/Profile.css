.my-profile {
  font-family: Arial, Helvetica, sans-serif;
  margin-top: 20px;
}

.my-profile .container {
  display: flex;
  justify-content: space-between;
  gap: 10px;
}

table {
  width: 50%;
  border-collapse: collapse;
}

thead {
  font-size: 28px;
  font-weight: 700;
  height: 50px;
}

.profile-row {
  text-align: left;
  font-size: 20px;
  border: 1px solid lightgray;
  height: 50px;
  line-height: 50px;
  text-indent: 20px;
}
