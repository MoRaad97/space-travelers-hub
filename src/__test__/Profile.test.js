import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import Profile from '../Components/Profile';

it('Missions Component snapshot test', () => {
  const profileRender = TestRenderer.create(
    <Provider store={store}>
      <Profile />
    </Provider>,
  )
    .toJSON();
  expect(profileRender).toMatchSnapshot();
});
