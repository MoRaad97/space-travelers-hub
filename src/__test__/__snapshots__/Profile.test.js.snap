// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Component snapshot test 1`] = `
<section
  className="my-profile"
>
  <div
    className="container"
  >
    <table>
      <thead>
        My Missions
      </thead>
      <tbody />
    </table>
    <table>
      <thead>
        My Rockets
      </thead>
      <tbody />
    </table>
  </div>
</section>
`;
